lib/config_builder/model/trigger.rb

```diff
@@ -0,0 +1,29 @@
+# Vagrant SSH credential model
+#
+# @see http://docs.vagrantup.com/v2/vagrantfile/ssh_settings.html
+class ConfigBuilder::Model::Trigger < ConfigBuilder::Model::Base
+  # @!attribute [rw] trigger
+  #   @return [String] The name of the trigger (before, after, instead_of)
+  def_model_attribute :trigger
+
+  # @!attribute [rw] command
+  #   @return [String] The vagrant command
+  def_model_attribute :command
+
+  # @!attribute [rw] host
+  #   @return [Hash] The trigger options
+  def_model_attribute :options
+
+  # @!attribute [rw] script
+  #   @return [String] The script
+  def_model_attribute :script
+
+  def to_proc
+    script = attr(:script)
+    Proc.new do |config|
+      config.trigger.send(attr(:trigger).to_sym, attr(:command).to_sym, attr(:options)) do
+        eval script
```

Comment:
The vagrant trigger has a method called [run](https://github.com/emyl/vagrant-triggers#usage). Here's a usage example:
```
---
triggers:
  -
    trigger: before
    command: destroy
    options:
      :stdout: true
    script: 'run "script/destroy_agent.rb #{@machine.ssh_info[:host]}"'
```